include/circt/Dialect/SystemC/SystemCStatements.td

```diff
@@ -108,6 +108,32 @@ def InstanceDeclOp : SystemCOp<"instance.decl", [
   }];
 }
 
+def BindPortOp : SystemCOp<"instance.bind_port", [HasParent<"CtorOp">]> {
+  let summary = "Binds a port of a module instance to a channel.";
+  let description = [{
+    The ports of a submodule have to be bound to channels in a module further
+    up in the instance hierarchy (as opposed to `sc_export` where the channel
+    has to reside in the same module or a submodule). Therefore, a port can be
+    bound to either a signal declared by the `systemc.signal` operation or to
+    a port with matching direction (and thus bound to a channel further up in
+    the hierarchy).
+    More information on ports can be found in IEEE 1666-2011 §5.12.,
+    in particular IEEE 1666-2011 §5.12.7. is about port binding.
+    More information on predefined channels can be found in IEEE 1666-2011 §6.
+  }];
+
+  let arguments = (ins ModuleType:$instance,
+                       StrAttr:$portName,
+                       ChannelType:$channel);
+
+  let assemblyFormat = [{
+    $instance `[` $portName `]` `to` $channel attr-dict `:`
```

Comment:
Alternatively, this could use the port number instead of a string. We've gone back and forth on this with FIRRTL and subfield/sub index and are slowly working towards unifying around fieldID. You may consider immediately jumping to this, but changing the printing/parsing to use the field name.

Reply:
Yeah, that makes a lot of sense. I'll change it as part of this PR.

Reply:
@seldridge Do you want to take another look at the changes regarding that comment (the newly added commit)?

Reply:
Nah, this looks good to go. 👍